pkg/runtime/controller.go

```diff
@@ -331,11 +331,31 @@ const (
 	istioProtocol = "istio-protocol"
 )
 
+// buildRule builds runtime representation of rule based on match condition
+func buildRule(k store.Key, r *cpb.Rule, rt ResourceType) (*Rule, error) {
+	if r.Match != "" {
+		m, err := expr.ExtractEQMatches(r.Match)
+		if err != nil {
+			return nil, err
+		}
+		if v := m[ContextProtocolAttributeName]; v == ContextProtocolTCP {
+			rt.protocol = protocolTCP
+		}
+		// TODO process destination.service and destination.name
```

Comment:
This is a new feature... I think, consider removing the TODO. Someone might think it is denoting incomplete code